src/thor/bidirectional_astar.cc

```diff
@@ -549,11 +561,17 @@ inline bool BidirectionalAStar::ExpandReverseInner(GraphReader& graphreader,
   float sortcost =
       newcost.cost + astarheuristic_reverse_.Get(t2->get_node_ll(meta.edge->endnode()), dist);
 
+  // There is an edge case where we may encounter only_restrictions with edges being
+  // marked as not_thru.  Basically the only way to get in this area is via this edge
+  // based on the restriction, but it is also marked as not_thru. We set the not_thru
+  // to false here only if allow_restricted_thru is true.  This bool is only set to true
+  // on the 2nd pass in route_action.  See the gurka test allow_restricted_thru
+  bool thru = (allow_restricted_thru() ? false : (pred.not_thru_pruning() || !meta.edge->not_thru()));
```

Comment:
Why is the condition being checked different than the forward direction? Here we're switching off pruning without checking if the predecessor edge has restriction or not (i.e using `pred.restrictions()`).

Reply:
@mandeepsandhu unfortunately we can't.  I will explain when I fill in the details above.

Reply:
I would simply have a not_thru_pruning option that is true on the first pass and false on the 2nd pass. No need (in my opinion) to check for restrictions - just disable all not_thru pruning on the 2nd pass. This was mostly added as a performance optimization for cycling and pedestrian (where hierarchies are not useful).